tdee_macro_calc/main.py

```diff
@@ -1,81 +1,49 @@
-    # Macros - first hit the calorie goal, then protein, then fat, then fill rest with carbs
-        # Protein 4 cal per gram, .8-1.0g/lb for building muscle, 120g-160g/day is a good place to start
-            # Output: how many grams of protein you should eat (and as an FYI, how many calories that is)
-        # Fat 9 cal per gram, minimum .3g/lb
-            # Output: how many grams of fat you should eat (and as an FYI, how many calories that is)
-        # Carbs 4 cal per gram Essentially, take the calories eaten in protein and fat, subtract from your goal calories per day, and the rest is what you can eat in carbs. However, carbs/fat really doesn't matter, what is most important is protein.
-            # Output: how many grams of carbs you should eat (and as an FYI, how many calories that is)
-
-# JSON Data structure DRAFT
-data = {
-    "user": {
-        "current_weight": 176, # this will be updated with a rolling 7-day average
-        "current_tdee": 2400, # this will be updated with a max 4 week (?) average
-        "sex": "male", # this determines default of 3.2, if female can go to 4.5 I believe. Not sure if I want to actually include this as functionality right now or not
-        "target_calorie_intake": 1700,
-        "per_lb_calorie_deficit": 3.2, # use 3.2 as default but allow user to change, possibly. TODO look at study to confirm dropping this number works
-        "per_lb_protein": .8, # use .8 as default but allow user to change, and/or play with different numbers in memory
-        "per_lb_fat": .3 # use .3 as default but allow user to change, and/or play with different numbers in memory
-    },
-    "tdee": [ # TODO This could totally be an object, same with user
-        {
-            "date": "9/20/22", # TODO this should be a datetime object and need to handle type checking
-            "weight": 178.1,
-            "calories": 2531
-        },
-        {
-            "date": "9/21/22",
-            "weight": 177.3,
-            "calories": 1971
-        },
-        {
-            "date": "9/22/22",
-            "weight": 179.2,
-            "calories": 2391
-        }
-    ]
-}
-
-def add_to_file():
-    try:
-        with open("data.json") as f:
-            new_data = json.load(f)
-            print(new_data)
-    except:
-        print("error opening file")
+def check_data_file() -> dict:
+    """
+    Checks to see if the file data.json exists, and if not it creates it with default values.
+
+    Returns a dictionary of the user and tdee information in the file.
+    """
 
-def tdee_input():
+    # Gather current user's directory information and set up ~/.tdee/data.json
+    home = Path.home()
+    tdee_directory = home / ".tdee"
+    tdee_directory.mkdir(exist_ok=True)
+    tdee_data = tdee_directory / "data.json"
+
+    # Try opening the data.json file in the .tdee folder
+    try:
+        with tdee_data.open() as f:
+            data = json.load(f)
```

Comment:
Not necessary, but you can replace the `with` statement by using the `.read_text` method.
```
contents = tdee_data.read_text()
data = json.loads(contents)
```

At this point, it's just a matter of code styling preference. I tend to aim for less indenting 🙂

Reply:
Okay, cool! That makes sense, and (I think) also has the added benefit of taking full advantage of the libraries we import. Just rambling—so the mindset could be if we're going to the trouble of importing it and it includes additional functionality (beyond the original intent) that makes our lives easier, might as well lean on it, especially if it allows cleaner code.

Reply:
Yeah, that's generally how I approach it. Python is a language that prides itself on readability, and some APIs (like pathlib) push that to the next level. They hide a lot of the ugliness behind cleaner APIs. A lot of Python libraries are targeted at making the language cleaner looking—not necessarily adding efficiency or anything like that.